bytecomp/symtable.ml

```diff
@@ -24,13 +24,54 @@ open Cmo_format
 
 module String = Misc.Stdlib.String
 
+module Compunit = struct
+  type t = compunit
+  module Set = Set.Make(struct type nonrec t = t let compare = compare end)
+  module Map = Map.Make(struct type nonrec t = t let compare = compare end)
+end
+
+let builtin_values = Predef.builtin_values
+
+module Predef = struct
+  type t = predef
+  module Set = Set.Make(struct type nonrec t = t let compare = compare end)
+  module Map = Map.Make(struct type nonrec t = t let compare = compare end)
+end
+
+module Global = struct
+  type t =
+    | Glob_compunit of compunit
+    | Glob_predef of predef
+
+  let name = function
+    | Glob_compunit (Compunit cu) -> cu
+    | Glob_predef (Predef_exn exn) -> exn
+
+  let quote s = "`" ^ s ^ "'"
+
+  let description = function
+    | Glob_compunit (Compunit cu) -> "compilation unit " ^ (quote cu)
+    | Glob_predef (Predef_exn exn) -> "predefined exception " ^ (quote exn)
+
+  let global_of_ident id =
```

Comment:
I would propose to rather optimize for the qualified name: `Global.of_ident` rather than `Global.global_of_ident`.